clang/include/clang/Driver/Options.td

```diff
@@ -2922,6 +2922,11 @@ def fxray_selected_function_group :
   HelpText<"When using -fxray-function-groups, select which group of functions to instrument. Valid range is 0 to fxray-function-groups - 1">,
   MarshallingInfoInt<CodeGenOpts<"XRaySelectedFunctionGroup">, "0">;
 
+defm xray_enable_shared : BoolFOption<"xray-enable-shared",
```

Comment:
A more idiomatic name would be `-f[no-]xray-shared`.

Reply:
Good point. I adjusted the option name as suggested.